sdk/armcore/policy_register_rp_test.go

```diff
@@ -316,3 +316,47 @@ func TestRPRegistrationPolicyCanCancel(t *testing.T) {
 		t.Fatalf("unexpected status code %d", resp.StatusCode)
 	}
 }
+
+func TestRPRegistrationPolicyDisabled(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	// initial response that RP is unregistered
+	srv.AppendResponse(mock.WithStatusCode(http.StatusConflict), mock.WithBody([]byte(rpUnregisteredResp)))
+	// polling responses to Register() and Get(), in progress
+	srv.RepeatResponse(5, mock.WithStatusCode(http.StatusOK), mock.WithBody([]byte(rpRegisteringResp)))
+	// polling response, successful registration
+	srv.AppendResponse(mock.WithStatusCode(http.StatusOK), mock.WithBody([]byte(rpRegisteredResp)))
```

Comment:
I think this is extra you already appended 5 responses to the mock server, although I think you could just append 3 for the max attempts.

Reply:
Good point, removed extra responses.